src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx

```diff
@@ -1,138 +1,229 @@
+import { useSharedState } from '../../../utils/useSharedState'
+import IndexStore from '../../index.store'
+import { getManifestResource } from './nodeDetail.api'
+import { showError } from '../../../../common'
+import MessageUI, { MsgUIType } from '../../../common/message.ui'
+import './nodeDetail.css'
+
+function NodeDetailComponent({
+    loadingResources,
+    isResourceBrowserView,
+    selectedResource,
+    logSearchTerms,
+    setLogSearchTerms,
+}: NodeDetailPropsType) {
     const [applicationObjectTabs] = useSharedState(
         AppDetailsStore.getAppDetailsTabs(),
         AppDetailsStore.getAppDetailsTabsObservable(),
-    );
+    )
     const appDetails = IndexStore.getAppDetails()
-    const params = useParams<{ actionName: string; podName: string; nodeType: string }>();
-    const [tabs, setTabs] = useState([]);
-    const [selectedTabName, setSelectedTabName] = useState('');
-    const { path, url } = useRouteMatch();
-    const history = useHistory();
+    const params = useParams<{ actionName: string; podName: string; nodeType: string; node: string }>()
+    const [tabs, setTabs] = useState([])
+    const [selectedTabName, setSelectedTabName] = useState('')
+    const [resourceContainers, setResourceContainers] = useState([])
+    const [isResourceDeleted, setResourceDeleted] = useState(false)
+    const [fetchingResource, setFetchingResource] = useState(false)
+    const { path, url } = useRouteMatch()
+
+    useEffect(() => {
+        const _tabs = getNodeDetailTabs(params.nodeType as NodeType)
+        setTabs(_tabs)
+    }, [params.nodeType])
 
     useEffect(() => {
-        const _tabs = getNodeDetailTabs(params.nodeType as NodeType);
-        setTabs(_tabs);
-    }, [params.nodeType]);
+        if (isResourceBrowserView && !loadingResources && selectedResource && params.node) {
+            getContainersFromManifest()
+        }
+    }, [loadingResources, params.node])
+
+    const getContainersFromManifest = async () => {
+        try {
+            setFetchingResource(true)
+            const { result } = await getManifestResource(
+                appDetails,
+                params.podName,
+                params.nodeType,
+                isResourceBrowserView,
+                selectedResource,
+            )
+
+            const _resourceContainers = []
+            if (result?.manifest?.spec) {
+                if (Array.isArray(result.manifest.spec.containers)) {
+                    _resourceContainers.push(...result.manifest.spec.containers.map((_container) => _container.name))
+                }
+
+                if (Array.isArray(result.manifest.spec.initContainers)) {
+                    _resourceContainers.push(...result.manifest.spec.containers.map((_container) => _container.name))
+                }
+            }
+            setResourceContainers(_resourceContainers)
+
+            // Clear out error on node change
+            if (isResourceDeleted) {
```

Comment:
should we do this in error case as well?

Reply:
done